### PR TITLE
Fixes #17480; tabling not breaking glass tables

### LINKED ISSUE #17480: Tabling somebody on a glass table does not break it
And I don't mean a holodeck table.

### PR DESCRIPTION
**Pull request by coiax**
Fixes #17480.

:cl: coiax
fix: Glass tables now break when people are pushed onto them.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -264,21 +264,38 @@
 	canSmoothWith = null
 	health = 50
 
+/obj/structure/table/glass/New()
+	. = ..()
+	SSobj.processing += src
```
Comment (KorPhaeron) [inline code/game/objects/structures/tables_racks.dm]: aaaaaaa

Comment (KorPhaeron) [inline code/game/objects/structures/tables_racks.dm]: this is not okay

Comment (RemieRichards) [inline code/game/objects/structures/tables_racks.dm]: CEASE AND DESIST.

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -264,21 +264,38 @@
...
 		return
 	// Don't break if they're just flying past
 	if(AM.throwing)
 		return
+	// If they land on this table, it'll break when process() is called
+
+	if(isliving(AM))
+		check_break(AM)
+
+/obj/structure/table/glass/process()
```
Comment (RemieRichards) [inline code/game/objects/structures/tables_racks.dm]: NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

File: code/game/objects/structures/tables_racks.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/objects/structures/tables_racks.dm]: DO NOT DO THIS I WILL KILL YOU

Comment (RemieRichards) [inline code/game/objects/structures/tables_racks.dm]: eyebrows

**Comment by optimumtact** [comment]
maybe not the best idea

File: code/game/objects/structures/tables_racks.dm [hunk 3]
```diff
@@ -263,29 +263,54 @@
...
 /obj/structure/table/glass/Crossed(atom/movable/AM)
 	. = ..()
 	if(flags & NODECONSTRUCT)
 		return
+	if(!isliving(AM))
+		return
 	// Don't break if they're just flying past
 	if(AM.throwing)
-		return
+		spawn(5)
```
Comment (optimumtact) [inline code/game/objects/structures/tables_racks.dm]: Is there no way to check that this is the last step of a throw without this spawn?

Like, modify the throwing proc so taht when it unsets AM.throwing it calls throw_impact on everything in the tile.

**Comment by KorPhaeron** [comment]
Why not just let glass tables break if you throw someone through them

File: code/game/objects/structures/tables_racks.dm (hunk 3 above)
Comment (coiax) [inline code/game/objects/structures/tables_racks.dm]: AM.throwing can be set to false in at least three different places.

1) When an atom Bumps another
2) When the throw hits something early
3) When the thrown item hits the turf of its target

I don't understand throw_impact enough to be able to change it. The side effects of being thrown onto a tile full of broken glass being painful is a really good emergent property, but I don't feel comfortable tackling that in this PR. Spawn will have to stay.

File: code/game/objects/structures/tables_racks.dm [hunk 4]
```diff
@@ -263,29 +263,54 @@
...
+			// Check again in a bit though
+			if(AM.loc == get_turf(src))
```
Comment (phil235) [inline code/game/objects/structures/tables_racks.dm]: AM could be null after the wait.

File: code/game/objects/structures/tables_racks.dm [hunk 5]
```diff
@@ -263,29 +263,54 @@
...
+	for(var/i in debris)
+		qdel(i)
+	. = ..()
 
 /obj/structure/table/glass/Crossed(atom/movable/AM)
 	. = ..()
 	if(flags & NODECONSTRUCT)
 		return
+	if(!isliving(AM))
+		return
```
Comment (phil235) [inline code/game/objects/structures/tables_racks.dm]: "var/mob/living/L = AM" after that would be nice.

**Pull request merged**